src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FFIDMetadataRepositorySpec.scala

```diff
@@ -58,16 +58,173 @@ class FFIDMetadataRepositorySpec extends AnyFlatSpec with TestDatabase with Scal
+    TestUtils.createFile(UUID.fromString(fileThreeId), consignmentId)
+    TestUtils.createFile(UUID.fromString(fileFourId), consignmentId)
+
+    val fileOneFfidMetadataId = TestUtils.addFFIDMetadata(fileOneId)
+    TestUtils.addFFIDMetadataMatches(fileOneFfidMetadataId.toString)
+
+    val fileTwoFfidMetadataId = TestUtils.addFFIDMetadata(fileTwoId)
+    TestUtils.addFFIDMetadataMatches(fileTwoFfidMetadataId.toString)
+
+    val fileThreeFfidMetadataId = TestUtils.addFFIDMetadata(fileThreeId)
+    TestUtils.addFFIDMetadataMatches(fileThreeFfidMetadataId.toString)
+
+    val ffidMetadataRows = ffidMetadataRepository.getFFIDMetadata(consignmentId).futureValue
+
+    ffidMetadataRows.length shouldBe 3
+  }
+
+  "getFFIDMetadata" should "return 0 ffid metadata rows for a consignment if no files were added" in {
+    val db = DbConnection.db
+    val ffidMetadataRepository = new FFIDMetadataRepository(db)
+    val consignmentId = UUID.fromString("21061d4d-ed73-485f-b433-c48b0868fffb")
+
+    TestUtils.createConsignment(consignmentId, userId)
+
+    val ffidMetadataRows = ffidMetadataRepository.getFFIDMetadata(consignmentId).futureValue
+
+    ffidMetadataRows.length shouldBe 0
+  }
+
+  "getFFIDMetadata" should "return the fileIds of the files that belong to that consignment" in {
+    val db = DbConnection.db
+    val ffidMetadataRepository = new FFIDMetadataRepository(db)
+    val consignmentOneId = UUID.fromString("21061d4d-ed73-485f-b433-c48b0868fffb")
+    val consignmentTwoId = UUID.fromString("89bdd66f-88c2-4edb-9911-2c605d002a1e")
+
+    val fileOneId = "50f4f290-cdcc-4a0f-bd26-3bb40f320b71"
+    val fileTwoId = "7f55565e-bfa5-4cf9-9e02-7e75ff033b3b"
+    val fileThreeId = "89b6183d-e761-4af9-9e37-2ffa09922dba"
+    val fileFourId = "939999e5-c082-4052-a99a-6b45092d826f"
+
+    TestUtils.createConsignment(consignmentOneId, userId)
+    TestUtils.createFile(UUID.fromString(fileOneId), consignmentOneId)
+    TestUtils.createFile(UUID.fromString(fileTwoId), consignmentOneId)
+    TestUtils.createFile(UUID.fromString(fileThreeId), consignmentOneId)
+    TestUtils.createFile(UUID.fromString(fileFourId), consignmentTwoId)
+
+    val fileOneFfidMetadataId = TestUtils.addFFIDMetadata(fileOneId)
+    TestUtils.addFFIDMetadataMatches(fileOneFfidMetadataId.toString)
+
+    val fileTwoFfidMetadataId = TestUtils.addFFIDMetadata(fileTwoId)
+    TestUtils.addFFIDMetadataMatches(fileTwoFfidMetadataId.toString)
+
+    val fileThreeFfidMetadataId = TestUtils.addFFIDMetadata(fileThreeId)
+    TestUtils.addFFIDMetadataMatches(fileThreeFfidMetadataId.toString)
+
+    val fileFourFfidMetadataId = TestUtils.addFFIDMetadata(fileFourId)
+    TestUtils.addFFIDMetadataMatches(fileFourFfidMetadataId.toString)
+
+    val ffidMetadataRows = ffidMetadataRepository.getFFIDMetadata(consignmentOneId).futureValue
+    val fileIds: Set[UUID] = ffidMetadataRows.toMap.keySet
+
+    fileIds.contains(UUID.fromString(fileOneId)) should equal(true)
```

Comment:
Minor thing, but these assertions could be written as:

```
fileIds should contain(UUID.fromString(fileOneId))
```

(Or `should not contain(foo)` in the final case.)

It means that if it fails, you get an error message like:

```
Set(50f4f290-cdcc-4a0f-bd26-3bb40f320b71, 7f55565e-bfa5-4cf9-9e02-7e75ff033b3b, 89b6183d-e761-4af9-9e37-2ffa09922dba) did not contain element 939999e5-c082-4052-a99a-6b45092d826f
```

Rather than:

```
false did not equal true
```

The former is easier to debug because you have more information about what the assertion was looking for.

It's a good principle to follow in general: if a test assertion is "something should equal true" (or false), then you can usually find a better assertion in the testing library's list of assertions. The main exception is if the function you're testing returns a simple boolean, because then the only thing you _can_ test is whether the result is true or false.

Reply:
Same goes for the next test as well.

Reply:
Ah, cool, I didn't know about this; thanks for tip. It also reads better/more naturally 👍